packages/agoric-cli/src/install.js

```diff
@@ -120,93 +235,44 @@ export default async function installMain(progname, rawArgs, powers, opts) {
-        if (!packageJSON) {
-          return;
-        }
-        const pj = JSON.parse(packageJSON);
-        for (const section of ['dependencies', 'devDependencies']) {
-          const deps = pj[section];
-          if (deps) {
-            for (const pkg of Object.keys(deps)) {
-              if (versions.has(pkg)) {
-                deps[pkg] = forceSdkVersion;
-              }
-            }
-          }
-        }
-        log.info(`updating ${pjson}`);
-        await fs.writeFile(pjson, `${JSON.stringify(pj, null, 2)}\n`);
       }),
     );
   }
 
-  const yarnInstall = await pspawn('yarn', [...linkFlags, 'install'], {
-    stdio: 'inherit',
-  });
-  if (yarnInstall) {
-    // Try to install via Yarn.
-    log.error('Cannot yarn install');
-    return 1;
+  if (forceSdkVersion !== undefined) {
+    // We need to update the SDK version in the package.jsons and `yarn.lock`.
+    await updateSdkVersion(forceSdkVersion);
   }
 
-  // Try to install via Yarn.
-  const yarnInstallUi = await (subdirs.includes('ui') &&
-    pspawn('yarn', [...linkFlags, 'install'], {
-      stdio: 'inherit',
-      cwd: 'ui',
-    }));
-  if (yarnInstallUi) {
-    log.error('Cannot yarn install in ui directory');
-    return 1;
-  }
+  await doYarnInstalls('updates');
 
-  if (sdkWorktree) {
+  if (sdkWorktree || !opts.sdk) {
+    log.info(chalk.bold.green('Done installing without SDK links'));
     return 0;
   }
 
-  if (opts.sdk) {
-    await Promise.all(
-      dirPackages.map(async ([dir, pjName]) => {
-        const SUBOPTIMAL = false;
-        if (SUBOPTIMAL) {
-          // This use of yarn is noisy and slow.
-          await pspawn('yarn', [...linkFlags, 'unlink', pjName]);
-          return pspawn('yarn', [...linkFlags, 'link'], {
-            stdio: 'inherit',
-            cwd: dir,
-          });
-        }
+  // Create symlinks to the SDK packages.
+  await Promise.all(
+    dirPackages.map(async ([dir, pjName]) => {
+      const SUBOPTIMAL = false;
+      if (SUBOPTIMAL) {
+        // This use of yarn is noisy and slow.
+        await pspawn('yarn', [...linkFlags, 'unlink', pjName]);
+        return pspawn('yarn', [...linkFlags, 'link'], {
+          stdio: 'inherit',
+          cwd: dir,
+        });
+      }
 
-        // This open-coding of the above yarn command is quiet and fast.
-        const linkName = `${linkFolder}/${pjName}`;
-        const linkDir = path.dirname(linkName);
-        log('linking', linkName);
-        return fs
-          .mkdir(linkDir, { recursive: true })
-          .then(_ => fs.symlink(path.relative(linkDir, dir), linkName));
-      }),
-    );
-  } else {
-    // Delete all old node_modules.
-    await Promise.all(
-      subdirs.map(subdir => {
-        const nm = `${subdir}/node_modules`;
-        log(chalk.bold.green(`removing ${nm}`));
-        return rimraf(nm);
-      }),
-    );
-  }
+      // This open-coding of the above yarn command is quiet and fast.
+      const linkName = `${linkFolder}/${pjName}`;
```

Comment:
Unless portability to older versions or Windows failures demand otherwise, I just use slashes and trust that Node.js's `fs` module figures out the rest.